src/Service.Tests/SqlTests/RestApiTests/Patch/PatchApiTestBase.cs

```diff
@@ -458,10 +449,259 @@ public virtual async Task PatchOneInsertWithDatabasePolicy()
+                    clientRoleHeader: "test_role_with_noread",
+                    expectedLocationHeader: string.Empty
+                );
+        }
+
+        /// <summary>
+        /// Tests that for a PATCH API request that results in a successful insert operation,
+        /// the response returned takes into account the include and exclude fields configured for the read action.
+        /// URI Path: Contains a non-existent PK.
+        /// Req Body: Valid Parameter with intended update.
+        /// Expects:
+        /// Status: 201 Created as PATCH results in an insert operation.
+        /// Response Body: Does not contain the categoryName field as it is excluded in the read configuration.
+        /// </summary>
+        [TestMethod]
+        public virtual async Task Patch_Insert_WithExcludeFieldsTest()
+        {
+            string requestBody = @"
+            {
+                ""piecesAvailable"": 4,
+                ""categoryName"": ""SciFi"",
+                ""piecesRequired"": 4
+            }";
+
+            await SetupAndRunRestApiTest(
+                   primaryKeyRoute: "categoryid/0/pieceid/7",
+                   queryString: null,
+                   entityNameOrPath: _Composite_NonAutoGenPK_EntityPath,
+                   sqlQuery: GetQuery("Patch_Insert_WithExcludeFieldsTest"),
+                   operationType: EntityActionOperation.UpsertIncremental,
+                   requestBody: requestBody,
+                   expectedStatusCode: HttpStatusCode.Created,
+                   clientRoleHeader: "test_role_with_excluded_fields",
+                   expectedLocationHeader: string.Empty
+               );
+        }
+
+        /// <summary>
+        /// Tests that for a PATCH API request that results in a successful insert operation,
+        /// the response returned takes into account the database policy configured for the read action.
+        /// URI Path: Contains a non-existent PK.
+        /// Req Body: Valid Parameter with intended update.
+        /// Expects:
+        /// Status: 201 Created as PATCH results in an insert operation.
+        /// Response Body: Empty. The database policy configured for the read action states that piecesAvailable cannot be 0.
+        /// Since, the PATCH request inserts a record with piecesAvailable = 0, the response must be empty.
+        /// </summary>
+        [TestMethod]
+        public virtual async Task Patch_Insert_WithReadDatabasePolicyTest()
+        {
+            string requestBody = @"
+            {
+                ""piecesAvailable"": 0,
+                ""categoryName"": ""SciFi"",
+                ""piecesRequired"": 4
+            }";
+
+            await SetupAndRunRestApiTest(
+                   primaryKeyRoute: "categoryid/0/pieceid/7",
+                   queryString: null,
+                   entityNameOrPath: _Composite_NonAutoGenPK_EntityPath,
+                   sqlQuery: GetQuery("PatchInsert_NoReadTest"),
```

Comment:
If the response should be empty, why is this picking the query: `PatchInsert_NoReadTest`?

Reply:
`PatchInsert_NoReadTest` query will not return any results because it has a condition `0 = 1`.